primo-zip.py

```diff
@@ -0,0 +1,97 @@
+# importing required modules
+import shutil
+import fileinput
+import json
+import os
+
+
+def set_secrets_and_zip(secret_data, types_to_check, base_directory, directory, zip_name):
+
+    # var for file paths list
+    file_paths = []
+
+    # crawl the directory structure
+    for root, directories, files in os.walk(directory):
+        for filename in files:
+            if (filename.endswith(types_to_check)):
+                filepath = os.path.join(root, filename)
+                file_paths.append(filepath)
+
+                # find and replace each secret tag with the actual secret
+                cnt = 0
+                for i in secret_data['secret_details']:
+                    secret_name = str(
+                        secret_data['secret_details'][cnt]['secret_name'])
+                    secret_value = str(
+                        secret_data['secret_details'][cnt]['secret_value'])
+
+                    with fileinput.FileInput(filepath, inplace=True) as file:
+                        for line in file:
+                            print(line.replace(
+                                str(secret_name), secret_value), end='')
```

Comment:
@darrowcoucla Most of the code in `set_secrets_and_zip` is exactly duplicated in `clear_secrets` - looks like the differences are line 30-31 and 60-61 (names vs values), and the additional zipping step at line 35.

Instead, how about pulling the common code into one routine, maybe `swap_secrets_and_values`?  Pass the 3 parameters the 2 existing routines have in common, and a new parameter which controls the swapping (values -> secrets or secrets -> values).

Then at line 86 where you call `set_secrets_and_zip`, first call this new procedure to do the appropriate swap, then call what's on line 35 to do the zipping.

That would help give a clear separation of responsibilities: swapping secrets/values, and the unrelated step of creating the zip archive.

Reply:
Pulled code into one routine.